src/normalize/normalizer.rs

```diff
@@ -92,7 +93,8 @@ unsafe impl crate::util::Pod for BuildIdNote {}
 
 trait BuildIdReader: 'static {
     fn read_build_id_from_elf(path: &Path) -> Result<Option<Vec<u8>>>;
-    fn read_build_id(parser: &ElfParser) -> Result<Option<Vec<u8>>>;
+    fn read_build_id_from_section_name(parser: &ElfParser) -> Result<Option<Vec<u8>>>;
+    fn read_build_id_from_notes(parser: &ElfParser) -> Result<Option<Vec<u8>>>;
```

Comment:
I don't think these two methods should be part of the trait. The reason we have the trait is to work with both a path as well as an already instantiated `ElfParser`. The `ElfParser` variant should probably just make sure to do the right thing, i.e., try reading from notes first and then from name. No need to expose these functions in the trait, though.